pkg/clusterapi/controlplane.go

```diff
@@ -0,0 +1,142 @@
+package clusterapi
+
+import (
+	"context"
+
+	etcdv1 "github.com/mrajashree/etcdadm-controller/api/v1beta1"
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+)
+
+// Object represents a kubernetes API object
+type Object interface {
+	kubernetes.Object
+}
+
+// ObjectComparator returns true only if only both kubernetes Object's are identical
+// Most of the time, this only requires comparing the Spec field, but that can variate
+// from object to object
+type ObjectComparator[O Object] func(current, new O) bool
+
+// ObjectRetriever gets a kubernetes API object using the provided client
+// If the object doesn't exist, it returns a NotFound error
+type ObjectRetriever[O Object] func(ctx context.Context, client kubernetes.Client, name, namespace string) (O, error)
+
+// ControlPlane represents the provider-specific spec for a CAPI control plane using the kubeadm CP provider
```

Comment:
We also have other provider-specific objects consider as CP objects (resourceset, secrets, configmap, etc.). Where will they be if not here?

Reply:
Great question! We will compose around this struct, building on top. Example: https://github.com/aws/eks-anywhere/pull/3432/files#diff-44aae5984bb86210e8481b396a2b8270c54ca9f136e264b0095e89ce141f765cR37